pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/DataType.java

```diff
@@ -64,26 +62,27 @@ public enum DataType {
     public final boolean requiresEscaping;
 
     /**
-     * The set of {@link BasicType}s that map to this type. Multiple types can map to a single instance of this enum.
+     * The set of primitive type names that map to this type. Multiple types can map to a single instance of this enum.
+     * Note: these strings are not case-normalized.
      */
-    private final Set<BasicType> basicTypes;
+    private final Set<String> basicTypeNames;
```

Comment:
I was struggling a bit with this change because of compatibility. But it seems, that the Set was not exposed and only used internally.

So I think this change still doesn't change any API.

Reply:
+1